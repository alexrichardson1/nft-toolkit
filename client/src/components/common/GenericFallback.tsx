import { FC } from "react";

interface PropsT {
  error: boolean;
}

const GenericFallback: FC<PropsT> = (props) => {
  return <> {props.error ? <>404</> : props.children}</>;
};

export default GenericFallback;
