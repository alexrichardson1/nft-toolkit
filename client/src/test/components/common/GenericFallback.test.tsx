import GenericFallback from "components/common/GenericFallback";
import { mount } from "enzyme";

test("Generic fallback", () => {
  const tree = mount(<GenericFallback error />);
  expect(tree).toMatchSnapshot();
});
