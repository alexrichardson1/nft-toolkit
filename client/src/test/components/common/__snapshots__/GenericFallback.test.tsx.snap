// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generic fallback 1`] = `
<GenericFallback
  error={true}
>
   
  404
</GenericFallback>
`;
